File: tests/methodology/fixtures/mock_budget_internal.ts
/**
 * Fixture for tests/methodology/test_mock_budget_lint.py.
 *
 * Internal-class mock. Violates TDD-2 internal-mock rule (target not at boundary).
 * Severity: Important (not in seam allowlist).
 */
import { vi, it, describe } from 'vitest';

describe('UserService', () => {
  it('mocks an internal module', async () => {
    vi.mock('./services/user-service');
  });
});

File: tests/methodology/fixtures/mock_budget_internal.py
"""Fixture for tests/methodology/test_mock_budget_lint.py.

Internal-class mock. Violates TDD-2 internal-mock rule (target not at boundary).
Severity: Important (not in seam allowlist).
"""
from unittest.mock import patch


@patch("src.services.user_service.UserService")
def test_mocks_internal_class(mock_service):
    """Mocks internal class - TDD-2 violation; should flag Important."""
    assert mock_service is not None
